lib/diff.ml

```diff
@@ -51,6 +62,28 @@ let modtype_item ~loc ~typing_env ~name ~reference ~current =
   | exception Includemod.Error _ ->
       Some (Module { mname = name; mdiff = Modified Unsupported })
 
+let type_item ~typing_env ~name ~reference ~current =
+  match (reference, current) with
+  | None, None -> None
+  | Some (Typ (reference, _)), None ->
+      Some (Type { tname = name; tdiff = Removed reference })
+  | None, Some (Typ (current, _)) ->
+      Some (Type { tname = name; tdiff = Added current })
+  | Some (Typ (reference, refId)), Some (Typ (current, curId)) -> (
+      let type_coercion1 () =
+        Includecore.type_declarations ~equality:true ~loc:current.type_loc
+          typing_env ~mark:true name current (Pident curId) reference
```

Comment:
Have you tried setting `mark` and `equality` to false? From what I gather they are just used to modify the typing env to set type usage and we have no use for this.

Could you try setting `mark` to false and not setting equality (it's an optional argument) to see if that still works?